RCodes/NonBinaryModelCheck.R

```diff
@@ -0,0 +1,164 @@
+library(data.table)
+library(glmnet)
+library(ggplot2)
+library(tidyverse)
+library(caret)
+library("ordinalNet")
+library("ordinalForest")
+library(tidyverse)
+library(caret)
+library("datarium")
+library(ggplot2)
+library("Metrics")
+library("MLmetrics")
+
+LogLossBinary = function(actual, predicted, eps = 1e-15) {
+  predicted = pmin(pmax(predicted, eps), 1-eps) - (sum(actual * log(predicted) + (1 - actual) * log(1 - predicted))) / length(actual)
+}
+y_test <- as.numeric(y_test)
+ordnet.pred 
+LogLossBinary(y_test,ordnet.pred)
+
+combined.binary.input.data<- data.table::fread("/Users/akhilachowdarykolla/Documents/Coding/development/PredictiveModelling/coaching_cwis_aggregrate_districts.csv")
+head(combined.binary.input.data)
+fold.vec <- sample(rep(1:10,l=nrow(combined.binary.input.data)))
+combined.binary.input.data <- data.table::data.table(combined.binary.input.data,fold=factor(fold.vec))
+
+combined.binary.input.data$BinaryETL <- ""
+for(i in 1:nrow(combined.binary.input.data)) {
+  row <- combined.binary.input.data[i,]
+  etlScore <- row$ETL.AVERAGE
+  print(etlScore)
+  if (etlScore >= 0 && etlScore <= 2.8) {
+    combined.binary.input.data[i,]$BinaryETL <- 1
+  } else{
+    combined.binary.input.data[i,]$BinaryETL <- 2
```

Comment:
you don't need for loop here (in general if you have a for loop over rows/observations you can probably replace it with a vectorized version which is simpler and more efficient)
ETL.thresh <- 2.8
combined.binary.input.data[, BinaryETL := ifelse(0 <= ETL.AVERAGE & ETL.AVERAGE <= ETL.thresh, paste0("below", ETL.thresh), paste0("above", ETL.thresh)) ]